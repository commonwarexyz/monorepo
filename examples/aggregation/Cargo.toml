[package]
name = "commonware-aggregation"
edition = "2021"
publish = true
version = "0.0.1"
license = "MIT OR Apache-2.0"
description = "aggregate signatures from multiple contributors over the BN254 curve."
readme = "README.md"
homepage = "https://commonware.xyz"
repository = "https://github.com/commonwarexyz/monorepo/tree/main/examples/aggregation"
documentation = "https://docs.rs/commonware-aggregation"

[dependencies]
commonware-cryptography = { workspace = true }
commonware-macros = { workspace = true }
commonware-p2p = { workspace = true }
commonware-runtime = { workspace = true }
commonware-utils = { workspace = true }
bytes = { workspace = true }
rand = { workspace = true }
tracing = { workspace = true }
prost = { workspace = true }
futures = { workspace = true }
clap = { workspace = true }
tracing-subscriber = { workspace = true }
prometheus-client = { workspace = true }
governor = { workspace = true }
ark-bn254 = "0.5.0"
ark-ff = "0.5.0"
ark-ec = "0.5.0"
ark-serialize = "0.5.0"
eigen-crypto-bn254 = "0.1.3"
eigen-crypto-bls = "0.1.3"

[build-dependencies]
prost-build = { workspace = true }

[[bin]]
name = "commonware-aggregation"
bench = false
