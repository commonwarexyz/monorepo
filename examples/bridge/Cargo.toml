[package]
name = "commonware-bridge"
edition = "2021"
publish = true
version = "0.0.2"
license = "MIT OR Apache-2.0"
description = "Send succinct consensus certificates between two networks."
readme = "README.md"
homepage = "https://commonware.xyz"
repository = "https://github.com/commonwarexyz/monorepo/tree/main/examples/bridge"
documentation = "https://docs.rs/commonware-bridge"

[dependencies]
commonware-cryptography = { workspace = true }
commonware-macros = { workspace = true }
commonware-p2p = { workspace = true }
commonware-runtime = { workspace = true }
commonware-utils = { workspace = true }
commonware-consensus = { workspace = true }
commonware-storage = { workspace = true }
commonware-stream = { workspace = true }
bytes = { workspace = true }
rand = { workspace = true }
tracing = { workspace = true }
futures = { workspace = true }
clap = { workspace = true }
prometheus-client = { workspace = true }
governor = { workspace = true }
tracing-subscriber = { workspace = true, features = ["fmt", "json"] }
prost = { workspace = true }

[build-dependencies]
prost-build = { workspace = true }

[[bin]]
name = "validator"
path = "src/bin/validator.rs"
bench = false

[[bin]]
name = "indexer"
path = "src/bin/indexer.rs"
bench = false

[[bin]]
name = "dealer"
path = "src/bin/dealer.rs"
bench = false
