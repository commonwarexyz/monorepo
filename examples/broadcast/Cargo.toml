[package]
name = "commonware-broadcast-example"
edition = "2021"
publish = false # Changed to false, not intended for individual publishing
version = "0.0.50"
license = "MIT OR Apache-2.0"
description = "Commonware broadcast example"
readme = "README.md"
homepage = "https://commonware.xyz"
repository = "https://github.com/commonwarexyz/monorepo/tree/main/examples/broadcast"
# documentation = "https://docs.rs/commonware-broadcast-example" # Usually for published crates

# Adding an empty workspace table to explicitly mark it as standalone
[workspace]

[dependencies]
commonware-cryptography = { version = "0.0.50", path = "../../cryptography" }
commonware-macros = { version = "0.0.50", path = "../../macros" }
commonware-p2p = { version = "0.0.50", path = "../../p2p" }
commonware-runtime = { version = "0.0.50", path = "../../runtime" }
commonware-utils = { version = "0.0.50", path = "../../utils" }
commonware-broadcast = { version = "0.0.50", path = "../../broadcast" }
commonware-storage = { version = "0.0.50", path = "../../storage" }

bytes = "1.7.1"
rand = "0.8.5"
tracing = "0.1.41"
futures = "0.3.31"
clap = { version = "4.5.18", features = ["derive"] } # Assuming derive feature is needed for clap
prometheus-client = "0.22.3"
governor = "0.6.3"
tracing-subscriber = { version = "0.3.19", features = ["fmt", "json"] }
chrono = "0.4.39"
ratatui = "0.27.0"
crossterm = "0.28.1"
serde_json = "1.0.122"

[[bin]]
name = "commonware-broadcast-example"
bench = false
