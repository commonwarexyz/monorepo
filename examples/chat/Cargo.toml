[package]
name = "commonware-chat"
edition = "2021"
publish = true
version = "0.0.3"

[dependencies]
commonware-p2p = { path = "../../p2p" }
