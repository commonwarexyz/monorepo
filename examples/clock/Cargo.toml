[package]
name = "commonware-clock"
edition = "2021"
publish = true
version = "0.0.2"
license = "MIT OR Apache-2.0"
description = "TBD"
readme = "README.md"
homepage = "https://commonware.xyz"
repository = "https://github.com/commonwarexyz/monorepo/tree/main/examples/clock"
documentation = "https://docs.rs/commonware-clock"

[dependencies]
commonware-cryptography = { workspace = true }
commonware-macros = { workspace = true }
commonware-p2p = { workspace = true } 
commonware-runtime = { workspace = true }
commonware-utils = { workspace = true }
commonware-consensus = { workspace = true }
bytes = { workspace = true } 
rand = { workspace = true }
tracing = { workspace = true } 
futures = { workspace = true }
clap = { workspace = true }
tracing-subscriber = { workspace = true }
prometheus-client = { workspace = true }
governor = { workspace = true }
