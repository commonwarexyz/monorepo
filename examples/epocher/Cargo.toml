[package]
name = "commonware-epocher"
edition = "2021"
publish = true
version = "0.0.58"
license = "MIT OR Apache-2.0"
description = "Continuously simulate epochs with rotating validator sets seeded by previous epoch's final block."
readme = "README.md"
homepage = "https://commonware.xyz"
repository = "https://github.com/commonwarexyz/monorepo/tree/main/examples/epocher"
documentation = "https://docs.rs/commonware-epocher"

[dependencies]
commonware-cryptography = { workspace = true }
commonware-codec = { workspace = true }
commonware-consensus = { workspace = true }
commonware-broadcast = { workspace = true }
commonware-macros = { workspace = true }
commonware-p2p = { workspace = true }
commonware-resolver = { workspace = true, features = ["mocks"] }
commonware-runtime = { workspace = true }
commonware-utils = { workspace = true }
commonware-storage = { workspace = true }
tracing = { workspace = true }
tracing-subscriber = { workspace = true, features = ["fmt", "json"] }
rand = { workspace = true }
clap = { workspace = true }
governor = { workspace = true }
bytes = { workspace = true }
futures = { workspace = true }
axum = { workspace = true }
tokio = { workspace = true, features = ["full"] }
reqwest = { workspace = true }

[[bin]]
name = "commonware-epocher"
bench = false
path = "src/bin/validator.rs"

[[bin]]
name = "commonware-epocher-indexer"
bench = false
path = "src/bin/indexer.rs"
