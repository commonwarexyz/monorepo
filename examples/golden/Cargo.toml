[package]
name = "commonware-golden"
version.workspace = true
edition.workspace = true
license.workspace = true
homepage.workspace = true

[dependencies]
# bulletproofs = { version = "5.0.0", features = ["yoloproofs"] }
rand.workspace = true
bytes.workspace = true
serde.workspace = true
futures.workspace = true
tracing.workspace = true
thiserror.workspace = true
governor.workspace = true
rand_core.workspace = true
serde_json.workspace = true
commonware-p2p.workspace = true
commonware-codec.workspace = true
commonware-utils.workspace = true
commonware-macros.workspace = true
prometheus-client.workspace = true
commonware-runtime.workspace = true
commonware-storage.workspace = true
commonware-resolver.workspace = true
commonware-broadcast.workspace = true
commonware-consensus.workspace = true
commonware-cryptography.workspace = true
clap = { workspace = true, features = ["derive"] }
